packages/server/graphql/private/mutations/runScheduledJobs.ts

```diff
@@ -35,19 +36,24 @@ const processMeetingStageTimeLimits = async (
   })
 }
 
-const jobProcessors = {
-  MEETING_STAGE_TIME_LIMIT_END: processMeetingStageTimeLimits
-}
-
-export type ScheduledJobUnion = Parameters<ValueOf<typeof jobProcessors>>[0]
+export type ScheduledJobUnion = ScheduledJobMeetingStageTimeLimit | ScheduledJobOrganizationLock
 
-const processJob = async (job: ScheduledJobUnion, {dataLoader}: {dataLoader: DataLoaderWorker}) => {
+const processJob = async (job: ScheduledJobUnion, dataLoader: DataLoaderWorker) => {
   const r = await getRethink()
   const res = await r.table('ScheduledJob').get(job.id).delete().run()
   // prevent duplicates. after this point, we assume the job finishes to completion (ignores server crashes, etc.)
   if (res.deleted !== 1) return
-  const processor = jobProcessors[job.type]
-  processor(job, {dataLoader}).catch(console.log)
+
+  if (job.type === 'MEETING_STAGE_TIME_LIMIT_END') {
+    return processMeetingStageTimeLimits(
+      job as ScheduledJobMeetingStageTimeLimit,
+      dataLoader
+    ).catch(console.log)
+  } else if (job.type === 'LOCK_ORGANIZATION') {
+    return processLockOrganizationJob(job as ScheduledJobOrganizationLock, dataLoader).catch(
```

Comment:
+1 we can avoid the casting here by checking if `orgId` exists in the job:

```
  if ('orgId' in job) {
    return processLockOrganizationJob(job, dataLoader).catch(console.log)
  } else {
    return processMeetingStageTimeLimits(job, {
      dataLoader
    }).catch(console.log)
  }
```

Reply:
changed here #7637